.github/workflows/build_and_test.yaml

```diff
@@ -0,0 +1,67 @@
+# See also https://github.com/EVerest/everest-core/tree/main/.github/workflows
+# TODO: modify to reuse the above workflow to DRY up CI.
+
+name: Build and test libtimer
+on: 
+  pull_request: {}
+  push:
+  workflow_dispatch:
+    inputs:
+      runner:
+        description: Which runner to use
+        type: choice
+        default: 'ubuntu-22.04'
+        required: true
+        options:
+          - 'ubuntu-22.04'
+          - 'large-ubuntu-22.04-xxl'
+jobs:
+  lint:
+    name: Lint
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout libtimer
+        uses: actions/checkout@v3
+        with:
+          path: source
+      - name: Run clang-format
+        uses: everest/everest-ci/github-actions/run-clang-format@v1.0.0
+        with:
+          source-dir: source
+          extensions: hpp,cpp
+          exclude: cache
+  install_and_test:
+    name: Install and test
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+    runs-on: ${{ matrix.os }}
```

Comment:
> I had to revert this to my initial as it was failing the run. The matrix run is not particularly useful with the current setup because the compile and run step all occurs within a docker. Changing the runner has not real impact on the outcome.

Yes, my suggestion was failing, because it had still a typo.
Please use 
```
runs-on: ${{ inputs.runner || 'ubuntu-22.04' }}
```
or remove the input. Otherwise it is confusing since the input hasn't any effect at the moment